internal/bucket/lifecycle/filter.go

```diff
@@ -122,32 +150,61 @@ func (f Filter) Validate() error {
 	if f.IsEmpty() {
 		return errXMLNotWellFormed
 	}
-	// A Filter must have exactly one of Prefix, Tag, or And specified.
+	// A Filter must have exactly one of Prefix, Tag, ObjectSize{LessThan,GreaterThan} or And specified.
+	type predType uint8
+	const (
+		emptyPred predType = iota
+		andPred
+		tagPred
+		prefixPred
+		sizeLtPred
+		sizeGtPred
+	)
+	var predCount int
+	var pType predType
 	if !f.And.isEmpty() {
-		if f.Prefix.set {
-			return errInvalidFilter
-		}
-		if !f.Tag.IsEmpty() {
-			return errInvalidFilter
-		}
-		if err := f.And.Validate(); err != nil {
-			return err
-		}
+		pType = andPred
+		predCount++
 	}
 	if f.Prefix.set {
-		if !f.Tag.IsEmpty() {
-			return errInvalidFilter
-		}
+		pType = prefixPred
+		predCount++
 	}
 	if !f.Tag.IsEmpty() {
-		if f.Prefix.set {
-			return errInvalidFilter
+		pType = tagPred
+		predCount++
+	}
+	if f.ObjectSizeGreaterThan > 0 {
+		pType = sizeGtPred
+		predCount++
+	}
+	if f.ObjectSizeLessThan > 0 {
+		pType = sizeLtPred
+		predCount++
+	}
+	if predCount > 1 {
+		return errInvalidFilter
+	}
```

Comment:
So in AWS you can only provide one predicate? because then why would at https://github.com/minio/minio/pull/18500/files#diff-f52f08b71d7c7f0ea61a74df360f8e1ea7daf770de643a7108b351e5d31b861bR58 there would be a check for `predCount < 2` ?

Reply:
An ILM rule can have exactly only one of the following predicates at the level of `<Filter></Filter>` such as,
-  `Tag` (singular)
-  `Prefix`
-  `And` (combination of 2 or more predicates)
-  `ObjectSizeLessThan`
-  `ObjectSizeGreaterThan`

So in this case, `predCount > 1` is incorrect.

When a user wishes to combine one or more predicates using the logical AND operator, then they will use `<And>`. This tag allows users to combine **two** or more predicates, which is why we check for `predCount > 2`.

Reply:
Add this as a comment since this is unnecessarily complicated for coders and users :-)